src/prototype_creep_harvest.js

```diff
@@ -43,8 +43,28 @@ Creep.prototype.handleSourcer = function() {
 };
 
 Creep.prototype.spawnCarry = function() {
-  var energyThreshold = Game.rooms[this.memory.base].controller.level * config.sourcer.spawnCarryLevelMultiplier;
-  var waitTime = config.sourcer.spawnCarryWaitTime;
+  if (this.memory.wait > 0) {
+    this.memory.wait -= 1;
+    return false;
+  }
+
+  var foundKey;
+  for (let key of Object.keys(config.carry.sizes).sort(function(a, b) { return parseInt(a, 10) - parseInt(b, 10); })) {
+    if (Game.rooms[this.memory.base].energyCapacityAvailable < key) {
+      break;
+    }
+    foundKey = key;
+  }
+  let carryCapacity = config.carry.sizes[foundKey][0] * CARRY_CAPACITY;
+
+  var work_parts = 0;
+  for (var part_i in this.body) {
+    if (this.body[part_i].type === 'work') {
+      work_parts++;
+    }
+  }
```

Comment:
maybe even something like this will be better https://github.com/samogot/screeps/blob/master/dist/extendPrototypes.js#L34-L53